flax/linen/recurrent.py

```diff
@@ -497,3 +500,189 @@ def initialize_carry(rng, batch_dims, size, init_fn=zeros):
+      init_key: a random key to initialize the cell's carry.
+      time_axis: the axis corresponding to the time dimension.
+      num_feature_dims: the number of feature dimensions, defaults to 1.
+      stateful: if True, the RNN cell's state is store in a variable named `carry` 
+        in the `cache` collection.
+      return_sequences: if True (default), the output contains the full sequence
+        of time steps, otherwise only the last output step is returned.
+      return_state: if True, a (output, carry) tuple is returned, otherwise only
+        the output is returned. Defaults to False.
+      go_backwards: if True, the input sequence is processed backwards. Defaults to False.
+      initial_state: an optional initial `carry` for the RNN cell. If stateful is True,
+        this will override the current value of the `carry`.s
+      variable_axes: the variable collections that are scanned over, this argument will 
+        be passed to `nn.scan`.
+      variable_broadcast: specifies the broadcasted variable collections, this argument
+        will be passed to `nn.scan`.
+      variable_carry: specifies the variable collections that are carried through
+        the loop, this argument will be passed to `nn.scan`.
+      split_rngs: a mapping from PRNGSequenceFilter to bool specifying whether to
+        a PRNG collection should be split such that its values are different at each
+        step, or replicated such that its values remain the same at each step. This
+        argument will be passed to `nn.scan`.
+
+    **Input shape**
+    
+    `x` should be an ndarray of shape `(b0...m, t, d0...dn)`where `b` are the batch
+    dimension, `t` is the time dimension, and `d` are the feature dimensions. The time
+    dimension can be in any position, `time_axis` should specify the corresponding
+    axis, internally the input will be reshaped to `(t, b0...m, d0...dn)`. The feature
+    dimensions should be last (after the time axis is to the front) and are specified by
+    `num_feature_dims`. The batch dimensions will simply be the remaining dimensions that
+    are not the time axis or the feature dimensions.
+
+    **Output shape**
+
+    If `return_sequences` is True, the output will have the same as the input, else
+    the time dimension will be dropped when the last time step is selected.
+    If `return_state` is True, the output will be a tuple of the output and the
+    carry.
+    """
+
+    if num_feature_dims < 1:
+      raise ValueError(f'num_feature_dims must be >= 1, got: {num_feature_dims}')
+    elif num_feature_dims >= len(x.shape):
+      raise ValueError(
+        f'\'num_feature_dims\' dimensions must be smaller than ' 
+        f'number of dimensions, got num_feature_dims \'{num_feature_dims}\' '
+        f'for input of shape \'{x.shape}\''
+      )
+
+    num_batch_dims = x.ndim - (num_feature_dims + 1) # features dims + time dim
+
+    # move time dimension to the first axis
+    x = x.swapaxes(0, time_axis)
```

Comment:
nit: transforms.scan does already support passing time_axis as the in axis

Reply:
Agreed, will make the change.